From 7d734ad4bc7fa2b7e513e6ec25b0847629c70a86 Mon Sep 17 00:00:00 2001
From: Cheick TRAORE <cheick.traore-ext@st.com>
Date: Fri, 10 May 2024 16:33:21 +0200
Subject: [PATCH 092/120] gpio: stmfx: Use correct flag values

Set the correct value for open_drain and push_pull modes.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/u-boot.git 9cf793df2575ea8a727e34743b629f3f52eeedf6 gpio: stmfx: Use correct flag values]
Change-Id: I98873851283d87f75535122543bb731dcfc3471a
Signed-off-by: Cheick TRAORE <cheick.traore-ext@st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/u-boot/+/388222
Domain-Review: Patrice CHOTARD <patrice.chotard@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Patrice CHOTARD <patrice.chotard@foss.st.com>
---
 drivers/pinctrl/pinctrl-stmfx.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/pinctrl/pinctrl-stmfx.c b/drivers/pinctrl/pinctrl-stmfx.c
index 61f335c4eb1..b77ea7dbdf5 100644
--- a/drivers/pinctrl/pinctrl-stmfx.c
+++ b/drivers/pinctrl/pinctrl-stmfx.c
@@ -177,9 +177,9 @@ static int stmfx_gpio_set_flags(struct udevice *dev, unsigned int offset,
 		if (flags & GPIOD_OPEN_SOURCE)
 			return -ENOTSUPP;
 		if (flags & GPIOD_OPEN_DRAIN)
-			ret = stmfx_conf_set_type(dev, offset, 0);
-		else /* PUSH-PULL */
 			ret = stmfx_conf_set_type(dev, offset, 1);
+		else /* PUSH-PULL */
+			ret = stmfx_conf_set_type(dev, offset, 0);
 		if (ret)
 			return ret;
 		ret = stmfx_gpio_direction_output(dev, offset, value);
-- 
2.47.2

